Geometry/GEMGeometryBuilder/src/GEMGeometryBuilderFromCondDB.cc

```diff
@@ -37,146 +36,182 @@ GEMGeometryBuilderFromCondDB::build(const std::shared_ptr<GEMGeometry>& theGeome
 				    const RecoIdealGeometry& rgeo )
 {
   const std::vector<DetId>& detids( rgeo.detIds());
-  std::vector<GEMSuperChamber*> superChambers;
-
-  std::string name;
-  std::vector<double>::const_iterator tranStart;
-  std::vector<double>::const_iterator shapeStart;
-  std::vector<double>::const_iterator rotStart;
-  std::vector<std::string>::const_iterator strStart;
-
-  for( unsigned int id = 0; id < detids.size(); ++id )
-  {  
-    GEMDetId gemid( detids[id] );
-    GEMDetId chid( gemid.region(), gemid.ring(), gemid.station(),
-		   gemid.layer(), gemid.chamber(), 0 );
-
-    tranStart = rgeo.tranStart( id );
-    shapeStart = rgeo.shapeStart( id );
-    rotStart = rgeo.rotStart( id );
-    strStart = rgeo.strStart( id );
-    name = *( strStart );
-
-    Surface::PositionType pos(*(tranStart)/cm, *(tranStart+1)/cm, *(tranStart+2)/cm );
-    // CLHEP way
-    Surface::RotationType rot(*(rotStart+0), *(rotStart+1), *(rotStart+2),
-                              *(rotStart+3), *(rotStart+4), *(rotStart+5),
-                              *(rotStart+6), *(rotStart+7), *(rotStart+8));
+  std::map<GEMDetId, GEMSuperChamber*> superChambers;
```

Comment:
`unordered_map` may be faster here

Reply:
yep that's fine, just have to change to rawID instead of GEMDetId